src/mongo/db/pipeline/expression.cpp

```diff
@@ -1729,9 +1729,18 @@ intrusive_ptr<ExpressionObject> ExpressionObject::parse(
 }
 
 intrusive_ptr<Expression> ExpressionObject::optimize() {
+    size_t countConstants = 0;
     for (auto&& pair : _expressions) {
+        // Check if Expression Constant
+        if (dynamic_cast<ExpressionConstant*>(pair.second.get())) {
+            countConstants++;
+        }
         pair.second = pair.second->optimize();
     }
+    // If all values in ExpressionObject are constant evaluate to ExpressionConstant
```

Comment:
End comments in a period. I also found this slightly hard to read - how about 
`// If all values in the object are constant, we can replace this ExpressionObject with a constant value.`